tests/phpunit/tests/functions/moveDir.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+/**
+ * Tests move_dir().
+ *
+ * @group functions.php
+ * @covers ::move_dir
+ */
+class Tests_Functions_MoveDir extends WP_UnitTestCase {
+
+	/**
+	 * Filesystem backup.
+	 *
+	 * @var null|WP_Filesystem_Base $wp_filesystem_backup
+	 */
+	private static $wp_filesystem_backup;
+
+	/**
+	 * The test directory.
+	 *
+	 * @var string $test_dir
+	 */
+	private static $test_dir;
+
+	/**
+	 * The existing 'from' directory path.
+	 *
+	 * @var string $existing_from
+	 */
+	private static $existing_from;
+
+	/**
+	 * The existing 'from' sub-directory path.
+	 *
+	 * @var string $existing_from_subdir
+	 */
+	private static $existing_from_subdir;
+
+	/**
+	 * The existing 'from' file path.
+	 *
+	 * @var string $existing_from_file
+	 */
+	private static $existing_from_file;
+
+	/**
+	 * The existing 'from' sub-directory file path.
+	 *
+	 * @var string $existing_from_subdir_file
+	 */
+	private static $existing_from_subdir_file;
+
+	/**
+	 * The existing 'to' directory file path.
+	 *
+	 * @var string $existing_to
+	 */
+	private static $existing_to;
+
+	/**
+	 * The existing 'to' file path.
+	 *
+	 * @var string $existing_to_file
+	 */
+	private static $existing_to_file;
+
+	/**
+	 * Sets up the filesystem and directory structure properties
+	 * before any tests run.
+	 */
+	public static function set_up_before_class() {
+		global $wp_filesystem;
+
+		self::$wp_filesystem_backup = $wp_filesystem;
+
+		parent::set_up_before_class();
+
+		/*
+		 * WP_Filesystem_MockFS has a bug that appears in CI.
+		 *
+		 * Until this is resolved, use WP_Filesystem_Direct and restore
+		 * $wp_filesystem to its original state after the tests.
+		 */
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		WP_Filesystem( array( 'method' => 'direct' ) );
+
+		self::$test_dir                  = trailingslashit( DIR_TESTDATA ) . 'functions/move_dir/';
```

Comment:
Can we use a temp directory via `wp_tempnam()`?

Reply:
I think traditionally all test directories/files use `DIR_TESTDATA` to set up resources for the tests.

Reply:
Some discussion between myself and @peterwilsoncc in DM showed that `get_temp_dir()` may be appropriate here, but that it might be a little awkward to ensure that a unique name exists. `wp_tempnam()` is for files, not directories, and may not work here.

If we have more thoughts on this, or if others do, please drop a comment. 🙂

Reply:
Myself and Peter decided on `get_temp_dir() . 'move_dir/'` which should work fine here.